source/pdk/1.0/pdk_get_started.md

```diff
@@ -0,0 +1,260 @@
+
+   `pdk new module module_name --skip-interview`
+
+1. Respond to the PDK dialog questions in the terminal. To accept the default value for any question, hit **Enter**.
+
+1. If the metadata that PDK displays is correct, confirm with `Y` to generate the module. If it is incorrect, enter `n` to cancel and start over.
+
+Related topics:
+
+* Install PDK
+* Add a new resource provider with PDK
+
+{:.concept}
+## Generating a class
+
+Generate classes for your module on the command line with PDK.
+
+To create a new class, run the `pdk new class` command, specifying the name of your new class. To generate an `init.pp` class, the main class of a module, generate a class with the same name as the module: `pdk new class module_name`.
+
+If the class name is not inside the module namespace, then the module name is automatically prepended to the class name. For example, if the module name is `apt` and the class name is `source`, the class is named `apt::source`.
+
+PDK creates the new class manifest and a test template file. You can then write tests in this template to validate your class's behavior.
+
+{:.concept}
+### Specifying class parameters
+
+If your new class should take parameters, specify them on the command line when you generate your class.
+
+Specify the values that parameter accepts, and optionally, specify the data type of the parameter with with the parameter name. You can provide any number of parameters on the command line.
+
+
+For example, to create a new class and define an `ensure` parameter for the class, run:
+
+``` bash
+pdk new class class_name "ensure:Enum['absent','present']"
+```
+
+This command creates a file in `module_name/manifests` named `class_name.pp`, with the ensure parameter defined. It also creates a test file in `module_name/spec/class` named `class_name_spec.rb`. This test file includes a basic template for writing your own unit tests.
+
+
+
+Related topics:
+
+* [Data types](TODO: LINK)
+
+{:.task}
+## Generate a new class
+
+To generate a new class in your module, use the `pdk new class` command.
+
+1. From the command line, in your module's directory, run:
+
+   ``` bash
+   pdk new class class_name "ensure:Enum['absent','present']"
+   ```
+
+{:.concept}
+## Testing your module with PDK
+
+PDK provides tools to help you run unit tests on your module and validate your new module's metadata, syntax, and style.
+
+By default, the PDK module template includes tools that can:
+
+* Validate the `metadata.json`.
+* Validate Puppet syntax.
+* Validate Puppet code style.
+* Validate Ruby code style.
+* Run unit tests.
+
+### Validating modules
+
+The validations included in PDK provide a basic check of the well-formedness of the module and syntax and style of the module's files. You do not need to write any tests for this validation.
+
+By default, the `pdk validate` command validates metadata, Ruby, and Puppet code syntax and style. To run a specific validation, add the name as an argument.
```

Comment:
Does ordering matter? Because validate will do metadata, Puppet, and then Ruby in that order, sans the parallel validation work.